src/dict.c

```diff
@@ -675,23 +694,36 @@ void dictRelease(dict *d)
 
 dictEntry *dictFind(dict *d, const void *key)
 {
-    dictEntry *he;
-    uint64_t h, idx, table;
+    dictEntry *he = NULL;
+    uint64_t h, idx;
 
     if (dictSize(d) == 0) return NULL; /* dict is empty */
-    if (dictIsRehashing(d)) _dictRehashStep(d);
+
     h = dictHashKey(d, key);
-    for (table = 0; table <= 1; table++) {
+    idx = h & DICTHT_SIZE_MASK(d->ht_size_exp[0]);
+    if ((long)idx >= d->rehashidx) {
+        he = d->ht_table[0][idx];
+    }
+
+    if (dictIsRehashing(d)) {
+        if ((long)idx >= d->rehashidx && he) {
```

Comment:
if `he` exists, `(long)idx >= d->rehashidx` should be true.
and the following `assert(!he);` could be ommited.
```suggestion
        if (he) {
```